### DISCUSSION TITLE
ダークモードでの `:::note alert` の色について

### DISCUSSION THREAD

**Discussion by yoshi389111** [increments/qiita-discussions#454] [要望・アイデア・不具合]
ベータ版のフィードバックにしたほうが良かったかもしれないのですが、画像の貼り付けなどができないのでこちらで失礼します。

ダークモードで、`:::note alert` の色合いについてですが、色覚多様性対応（カラーユニバーサルデザイン対応）的に、問題がありそうな気がしています。
より具体的には、コントラストが低いような気がしています。

Chrome の DevToolsで Rendering のエミュレートをしてみたところ以下のような感じでした（サイズがバラバラですみません）。
画像は、[Markdown記法 チートシート](https://qiita.com/Qiita/items/c686397e4a0f4f11683d)の「Note - 補足説明」の箇所です。

Protanopia（1型2色覚/赤色感受性欠如）の場合：

![image](https://user-images.githubusercontent.com/15214393/227833673-6b303b43-d379-4640-b022-6561aaff3476.png)

Deuteranopia（2型2色覚/緑色感受性欠如）の場合：

![image](https://user-images.githubusercontent.com/15214393/227833811-1fbcc7eb-2514-4587-9be4-9021964aadc3.png)

Tritanopia（3型2色覚/青色感受性欠如）の場合：

![image](https://user-images.githubusercontent.com/15214393/227833974-a2c54baf-db13-4274-866e-311f25875393.png)

Achromatopsia（1色覚）の場合：

![image](https://user-images.githubusercontent.com/15214393/227834463-3b873948-cfc5-45f3-b2c8-4b9f381fefe0.png)

素人判断ですが、Protanopia と Achromatopsia が、ずいぶんと視認性が低いように思います。

Protanopia については、[色覚の多様性と色覚バリアフリーなプレゼンテーションの第2回](https://www.nig.ac.jp/color/barrierfree/barrierfree2-2.html)によると

> PowerPoint やカラースライドで黒や紺、濃い青色の背景に「濃い赤」の文字があると、文字が目立たないどころか読むことさえできない（図6D）。「濃い赤」を使わずに、少しだけ白を混ぜた「明るい赤」（R、G、B ＝ 100％、20％、20％） や短波長 （オレンジ） 寄りの「朱色」 （R、G、B ＝ 100％、20％、0％） を使うと、同じ赤でも緑視物質や青視物質を刺激することが可能となり、黒い文字列の中でも強調され、暗い背景の中でも文字が読めるようになる （図6B、D）。

とのことなので、もう少し明るくするか、他の色を混ぜた方が良いのではないかと思います。

Achromatopsia に関しては、やはりコントラストをもう少し大きくするべきではないかと思います。

ベータテスターに Protanopia の方や、Achromatopsia の方がいらっしゃるのかわからないですが、Qiita 全体としては Protanopia の方はそれなりのユーザー数になるのではないかと思っています。
（日本人男性だと、Protanopia と Deuteranopia の方を合わせると 20 人に 1 人ぐらいだそうです。Protanopia の方だけの割合はわかりませんが）

**Comment by kskwtnk** [comment]
@yoshi389111 
ご意見ありがとうございます！
色の見え方について、非常にごもっともな指摘です

修正したいと思っていますが、時間はかかってしまいそう、というのが回答です :pray:
ダークモードに限らず現状のQiitaのカラーパレットにはアクセシビリティ的に問題がある箇所がいくつかあります
解消したいと思っているのですが、いかんせん範囲が広く少しずつしか着手できていません

ダークモード制作でも、本来であればダークモード用にパレットを別途用意する必要があります
ただ、そこまでスコープを広げると、今でも遅い提供がもっと遅くなる……というジレンマがありました
というわけで、今回は既存のパレットの範囲での提供にいたっています
